src/xhydro/frequency_analysis/uncertainties.py

```diff
@@ -213,38 +212,115 @@ def calc_q_iter(
     """
     # We select groups for all or one id
     if bv == "all":
-        ds_temp = ds_groups[[var]].dropna("group_id", how="all")
+        ds_temp = ds_groups.dropna("group_id", how="all")
     else:
-        ds_temp = ds_groups[[var]].sel(id=bv).dropna("group_id", how="all")
+        ds_temp = ds_groups.sel(id=bv).dropna("group_id", how="all")
     ds_mom = []
 
     # For each group, we find which id are in it
     for group_id in ds_temp.group_id.values:
         id_list = ds_groups.sel(group_id=group_id).dropna("id", how="all").id.values
         # We use moments with ressample previously done, and we create ds_moment_group with iterations
         ds_mom.append(
-            ds_moments_iter[[var]]
-            .sel(id=id_list)
+            ds_moments_iter.sel(id=id_list)
             .assign_coords(group_id=group_id)
             .expand_dims("group_id")
         )
 
     # Concat along group_id
     ds_moments_groups = xr.concat(ds_mom, dim="group_id")
-    ds_groups = (
-        ds_groups[[var]]
-        .sel(group_id=ds_moments_groups.group_id.values)
-        .dropna(dim="id", how="all")
+    ds_groups = ds_groups.sel(group_id=ds_moments_groups.group_id.values).dropna(
```

Comment:
```suggestion
    ds_groups = ds_groups.sel(group_id=ds_moments_groups.group_id).dropna(
```
Je ne pense pas que `values` soit nécessaire, vu que c'est une dimension ?